lib/src/Uncertainty/Distribution/EmpiricalBernsteinCopula.cxx

```diff
@@ -0,0 +1,428 @@
+    ++linearIndex;
+  }
+  return realization;
+}
+
+/* Get a sample of the distribution */
+Sample EmpiricalBernsteinCopula::getSample(const UnsignedInteger size) const
+{
+  const UnsignedInteger dimension = getDimension();
+  SampleImplementation sample(size, dimension);
+  const UnsignedInteger mixtureSize = copulaSample_.getSize();
+  for (UnsignedInteger i = 0; i < size; ++i)
+  {
+    // Select the atom
+    const UnsignedInteger atomIndex(RandomGenerator::IntegerGenerate(mixtureSize));
+    UnsignedInteger linearIndex = atomIndex * dimension;
+    for (UnsignedInteger j = 0; j < dimension; ++j)
+    {
+      sample(i, j) = DistFunc::rBeta(logFactors_[linearIndex], binNumber_ - logFactors_[linearIndex] + 1.0);
+      ++linearIndex;
+    } // j
+  } // i
+  sample.setDescription(getDescription());
+  return sample;
+}
+
+/* Get the PDF of the EmpiricalBernsteinCopula */
+Scalar EmpiricalBernsteinCopula::computePDF(const Point & point) const
+{
+  const UnsignedInteger dimension = getDimension();
+  if (point.getDimension() != dimension) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=" << dimension << ", here dimension=" << point.getDimension();
+  for (UnsignedInteger i = 0; i < dimension; ++i)
+    if ((point[i] <= 0.0) || (point[i] >= 1.0)) return 0.0;
+  Scalar pdfValue = 0.0;
+  Point logX(dimension);
+  Point log1pX(dimension);
+  for (UnsignedInteger i = 0; i < dimension; ++i)
+  {
+    logX[i] = std::log(point[i]);
+    log1pX[i] = log1p(-point[i]);
+  }
+  const UnsignedInteger size = copulaSample_.getSize();
+  UnsignedInteger linearIndex = 0;
+  for (UnsignedInteger i = 0; i < size; ++i)
+  {
+    Scalar logPDFAtom = 0.0;
+    for (UnsignedInteger j = 0; j < dimension; ++j)
+    {
+      logPDFAtom += (logFactors_[linearIndex] - 1.0) * logX[j] + (binNumber_ - logFactors_[linearIndex]) * log1pX[j] - logBetaFactors_[linearIndex];
+      ++linearIndex;
+    } // j
+    pdfValue += std::exp(logPDFAtom);
+  } // i
+  return pdfValue / size;
+}
+
+/* Get the log-PDF of the EmpiricalBernsteinCopula */
+Scalar EmpiricalBernsteinCopula::computeLogPDF(const Point & point) const
+{
+  const UnsignedInteger dimension = getDimension();
+  if (point.getDimension() != dimension) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=" << dimension << ", here dimension=" << point.getDimension();
+  for (UnsignedInteger i = 0; i < dimension; ++i)
+    if ((point[i] <= 0.0) || (point[i] >= 1.0)) return -SpecFunc::LogMaxScalar;
+  Scalar sumPDFValue = 0.0;
+  Point logX(dimension);
+  Point log1pX(dimension);
+  for (UnsignedInteger i = 0; i < dimension; ++i)
+  {
+    logX[i] = std::log(point[i]);
+    log1pX[i] = log1p(-point[i]);
+  }
+  const UnsignedInteger size = copulaSample_.getSize();
+  UnsignedInteger linearIndex = 0;
+  Scalar maxLogPDF = 0.0;
+  const Scalar logW = -std::log(size);
```

Comment:
error C2668: 'std::log' : ambiguous call to overloaded function
=> std::log(static_cast<Scalar>(size))